### PR TITLE
SSP 4.99.151

### PR DESCRIPTION
**Pull request by cPanelChrisD**

### CODE REVIEW

File: ssp [hunk 1]
```diff
@@ -1754,6 +1755,26 @@ sub print_cpanel_info {
     print_info('cPanel Info: ');
     print_normal($output);
 
+    my %eol = (
+        '56' => { expires => 1509408001, text => 'October 31, 2017' },
+        '58' => { expires => 1501459201, text => 'July 31, 2017' },
+        '60' => { expires => 1509408001, text => 'October 31, 2017' },
+        '62' => { expires => 1522454401, text => 'March 31, 2018' }
+    );
```
Comment (davehhpnet-zz) [inline ssp:1763]: Seems like this is - going to need to be constantly updated.  At least until we roll out TIERS.json in it's entirety.  Right now we only have 11.62 defined, afaik.
```
      "11.62" : [
         {
            "build" : "11.62.0.21",
            "expires" : 1522454401,
            "is_lts" : 1,
            "is_main" : 1,
            "named" : [
               "stable"
            ]
         }
      ],
```

But ultimately we do want this re-written to where we can just yank down that data, build the data structure on-the-fly, and never have to circle back to re-define all of these versions/expirations.

@cPanelChrisD for now I don't see a way around how this is but it's something to keep in mind for code portability later on.

File: ssp [hunk 2]
```diff
@@ -2728,6 +2749,64 @@ sub print_news {
...
+    }
+
+    if (
+        $license_error =~ m{ \A 
+         \QThe hostname must be a Fully Qualified Domain Name! (\E.+\)
+         \Z }xms
+      ) {
+        print_warn('License Error: ');
+        print_warning( '[ ' . $license_error . ' ]' );
+        print_warning('         \_ If this license error is not resolved after correctly setting the hostname, fork ticket for license issue if not related to current issue, send "ESCALATE - License issue to Dev" response, and escalate ticket to "QA/Development".');
```
Comment (davehhpnet-zz) [inline ssp:2786]:
```
"ESCALATE - License issue to Dev" response, and escalate ticket to "QA/Development"
```
We are reusing this text enough that we should store it in a $scalar

File: ssp [hunk 3]
```diff
@@ -2728,6 +2749,64 @@ sub print_news {
...
+         \QAbort, Retry, Fail?\E
+       | \QDoes not compute!\E
+       | \QReturn without Gosub.\E
+       | \QPrinting is not supported on this printer.\E
+       | \QCannot issue a license to \E[^ ]+\Q without a \E(DISTRO|OSVER)\.
+       \Z }xms
+      ) {
+        print_crit('License Error: ');
+        print_critical( '[ ' . $license_error . ' ]' );
+        print_critical('         \_ Fork ticket for license issue if not related to current issue, send "ESCALATE - License issue to Dev" response, and escalate ticket to "QA/Development".');
```
Comment (davehhpnet-zz) [inline ssp:2801]: Same reuse of string that should be in $scalar

File: ssp [hunk 4]
```diff
@@ -2933,7 +3012,7 @@ sub check_for_rawopts {
     }
 
     if (@dir_contents) {
-        print_warn('Rawopts Detected: ');
+        print_warn('EA3 Rawopts Detected: ');
```
Comment (davehhpnet-zz) [inline ssp:3015]: I like the clarifications here to help differentiate between EA versions.

File: ssp [hunk 5]
```diff
@@ -3070,13 +3150,13 @@ sub check_for_local_makecpphp_template {
     my $makecpphp_local_profile = '/var/cpanel/easy/apache/profile/makecpphp.profile.yaml.local';
 
     if ( -e $makecpphp_local_profile ) {
-        print_warn('makecpphp Local Profile: ');
+        print_warn('EA3 makecpphp Local Profile: ');
```
Comment (davehhpnet-zz) [inline ssp:3153]: Works but ideally we don't _have_ to have this as makecpphp is EA3 specific

File: ssp [hunk 6]
```diff
@@ -6361,7 +6449,7 @@ sub check_for_usr_local_include_jpeglib_h {
 }
 
 sub check_for_bw_module_and_more_than_1024_vhosts {
-    my $httpdconf = '/usr/local/apache/conf/httpd.conf';
+    my $httpdconf = $EA4 ? '/etc/apache2/conf/httpd.conf' : '/usr/local/apache/conf/httpd.conf';
```
Comment (davehhpnet-zz) [inline ssp:6452]: In either case /usr/local/apache/conf/httpd.conf will be a valid path here, right?  I suppose this is a better path than changing the -f on line 6365/6453 to a -e, though.  OK

**Review by davehhpnet-zz — APPROVED** [review]
Looks good to go!

**Comment by davelcpanelnet-zz** [comment]
My apologies I mixed up accounts when doing the review.  This looks good to go!

**Pull request merged**